engine-precompiles/src/alt_bn256.rs

```diff
@@ -67,6 +67,15 @@ mod consts {
     pub(super) const ERR_BIG_ENDIAN: ExitError = ExitError::Other(Borrowed("ERR_BIG_ENDIAN"));
 }
 
+#[cfg(feature = "contract")]
+mod type_arith {
+    pub struct Double<const P: usize>;
+    pub trait Is<const S: usize> {}
+    impl Is<128> for Double<64> {}
+    impl Is<64> for Double<32> {}
+    impl Is<32> for Double<16> {}
+}
```

Comment:
Whats the motivation?

Reply:
See https://github.com/aurora-is-near/aurora-engine/pull/719#discussion_r1128073168